File: c-003/main.r
largestPrimeFactor <- function(num) {
    for (i in 2:(num-1)) {
        if (num %% i == 0) {
            return(largestPrimeFactor(num / i))
        }
    }
    num
}

print(largestPrimeFactor(600851475143))

File: c-003/main.js
function largestPrimeFactor(num) {
	for (let i = 2; i < num; i++) {
        if (num % i === 0) {
            return largestPrimeFactor(num / i)
        }
    }
    return num
}

console.log(largestPrimeFactor(600_851_475_143))
